packages/app/src/sandbox/eval/transpiled-module.ts

```diff
@@ -565,160 +565,170 @@ export default class TranspiledModule {
-    }`;
+      const sourceEqualsCompiled = code === this.module.code;
+      const sourceURL = `//# sourceURL=${location.origin}${this.module.path}${
+        this.query ? `?${this.hash}` : ''
+      }`;
 
-    // Add the source of the file by default, this is important for source mapping
-    // errors back to their origin
-    code = `${code}\n${sourceURL}`;
+      // Add the source of the file by default, this is important for source mapping
+      // errors back to their origin
+      code = `${code}\n${sourceURL}`;
 
-    this.source = new ModuleSource(
-      this.module.path,
-      code,
-      finalSourceMap,
-      sourceEqualsCompiled
-    );
+      this.source = new ModuleSource(
+        this.module.path,
+        code,
+        finalSourceMap,
+        sourceEqualsCompiled
+      );
 
-    if (
-      this.previousSource &&
-      this.previousSource.compiledCode !== this.source.compiledCode
-    ) {
-      const hasHMR = manager.preset
-        .getLoaders(this.module, this.query)
-        .some(t =>
-          t.transpiler.HMREnabled == null ? true : t.transpiler.HMREnabled
-        );
+      if (
+        this.previousSource &&
+        this.previousSource.compiledCode !== this.source.compiledCode
+      ) {
+        const hasHMR = manager.preset
+          .getLoaders(this.module, this.query)
+          .some(t =>
+            t.transpiler.HMREnabled == null ? true : t.transpiler.HMREnabled
+          );
 
-      if (!hasHMR) {
-        manager.markHardReload();
-      } else {
-        this.resetCompilation();
+        if (!hasHMR) {
+          manager.markHardReload();
+        } else {
+          this.resetCompilation();
+        }
       }
-    }
 
-    await Promise.all(
-      this.asyncDependencies.map(async p => {
-        try {
-          const tModule = await p;
+      await Promise.all(
+        this.asyncDependencies.map(async p => {
+          try {
+            const tModule = await p;
 
-          this.dependencies.add(tModule);
-          tModule.initiators.add(this);
-        } catch (e) {
-          /* let this handle at evaluation */
-        }
-      })
-    );
+            this.dependencies.add(tModule);
+            tModule.initiators.add(this);
+          } catch (e) {
+            /* let this handle at evaluation */
+          }
+        })
+      );
 
-    this.asyncDependencies = [];
+      this.asyncDependencies = [];
 
-    await Promise.all(
-      flattenDeep([
-        ...Array.from(this.transpilationInitiators).map(t =>
-          t.transpile(manager)
-        ),
-        ...Array.from(this.dependencies).map(t => t.transpile(manager)),
-      ])
-    );
+      await Promise.all(
+        flattenDeep([
+          ...Array.from(this.transpilationInitiators).map(t =>
+            t.transpile(manager)
+          ),
+          ...Array.from(this.dependencies).map(t => t.transpile(manager)),
+        ])
+      );
 
-    return this;
+      return this;
+    } finally {
```

Comment:
This is really what I've added, when finished with transpiling we remove it from the transpileJobs.